tests/Knp/Menu/Tests/Matcher/Voter/RequestStackRouteVoterTest.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+namespace Knp\Menu\Tests\Matcher\Voter;
+
+use Knp\Menu\Matcher\Voter\RequestStackRouteVoter;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class RequestStackRouteVoterTest extends \PHPUnit_Framework_TestCase
+{
+    protected function setUp()
+    {
+        if (!class_exists('Symfony\Component\HttpFoundation\RequestStack')) {
+            $this->markTestSkipped('The Symfony HttpFoundation component is not available.');
+        }
+    }
+
+    public function testMatchingWithoutRequest()
+    {
+        $item = $this->getMock('Knp\Menu\ItemInterface');
+        $item->expects($this->never())
+            ->method('getExtra');
+
+        $voter = new RequestStackRouteVoter(new RequestStack());
+
+        $this->assertNull($voter->matchItem($item));
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testInvalidRouteConfig()
+    {
+        $item = $this->getMock('Knp\Menu\ItemInterface');
+        $item->expects($this->any())
+            ->method('getExtra')
+            ->with($this->logicalOr($this->equalTo('routes'), $this->equalTo('routesParameters')))
+            ->will($this->returnCallback(function ($parameter) {
+                switch ($parameter) {
+                    case 'routes':
+                        return array(array('invalid' => 'array'));
+                    case 'routesParameters':
+                        return array();
+                }
+            }));
+
+        $requestStack = new RequestStack();
+
+        $request = new Request();
+        $request->attributes->set('_route', 'foo');
+        $request->attributes->set('_route_params', array());
+
+        $requestStack->push($request);
```

Comment:
we should have tests with more than one request in the stack, that make sure the right element on the stack is used (current resp master)